### PR TITLE
Remove temporary TransportAction constructors

### LINKED ISSUE #97879: Make transportService.registerRequestHandler take an executor instead of its name
Similarly to https://github.com/elastic/elasticsearch/pull/97873 but for request handlers: let's stop passing around strings and looking these things up by name on each request when we could just pass around the `Executor` itself.

### PR DESCRIPTION
**Pull request by DiannaHohensee**
Temporary constructors were added for serverless compatibility
until serverless could be updated to use new constructors.

Closes #97879

### CODE REVIEW

**Comment by DiannaHohensee** [comment]
Unrelated failure, issue already filed [here](https://github.com/elastic/elasticsearch/issues/99913).

**Review by DaveCTurner — APPROVED** [review]
LGTM once the dependent PR is merged

**Comment by DiannaHohensee** [comment]
For future reference, related serverless PR: https://github.com/elastic/elasticsearch-serverless/pull/955

**Pull request merged**
